javascript/javascript3/week3/shopping-cart.js

```diff
@@ -0,0 +1,132 @@
+class Product {
+    constructor(name, price) {
+      this.name = name;
+      this.price = price;
+    }
+    convertToCurrency(currency){
+      
+      let price = 0
+      switch(currency){
+        case "dollars": {
+          price = this.price * 0.16
+          price = price + ' $'
+          break;
+        }
+        case "pounds":{
+          price = this.price * 0.12
+          price = price + ' £'
+          break;
+        }
+        case "euros": {
+          price = this.price * 0.13
+          price = price + ' €'
+          break;
+        }
+      }
+      return price
+    }
+  }
+  
+  class ShoppingCart {
+    constructor() {
+      this.products = [];
+    }
+  
+    addProduct(product) {
+      // Implement functionality here
+      this.products.push(product);
+    }
+  
+    removeProduct(product) {
+      // Implement functionality here
+      const removeItem = this.products.filter(item => item.name !== product.name)
+      this.products = removeItem;
+    }
+  
+    searchProduct(productName) {
+
+      // Implement functionality here
+      const searchProduct = this.products.filter(item => item.name === productName);
+      const trs = document.querySelectorAll(".data-row");
+      trs.forEach(tr => tr.remove());
+      this.renderProducts(searchProduct);
+      return searchProduct;
+    }
+  
+    getTotal() {
+      // Implement functionality here
+      const reducer = (accumulator, currentValue) => accumulator + currentValue;
+      const totalProductsPrice = this.products.map(product => product.price).reduce(reducer,0);
+      return totalProductsPrice;
+
+    }
+    renderProducts(products) {
+      const table = document.querySelector("table > tbody");
+      products.forEach(product => {
+        const tr = document.createElement("tr");
+        tr.setAttribute("class", "data-row");
+        const product_td = document.createElement("td");
+        product_td.innerText = product.name;
+        const price_td = document.createElement("td");
+        price_td.innerText = `${product.price} dkk`;
+        tr.appendChild(product_td);
+        tr.appendChild(price_td);
+        table.appendChild(tr);
+      })
+    }
+  
+    getUser() {
+      // Implement functionality here
+      return new Promise((resolve, reject)=>{
+        fetch("https://jsonplaceholder.typicode.com/users/1")
+        .then((res)=> res.json())
+        .then((data)=> {
+          this.renderProducts(this.products)
+          this.getTotal()
+          resolve(data)
+        console.log("promise resolved")})
+        .catch((e)=>reject(e));
+      })
```

Comment:
Again fetch returns a Promise by default so I don't see a point in wrapping it into a new Promise. You can just return fetch itself.

Reply:
I fixed it. if something else I still need to change. Please comment!!